### ISSUE TITLE
b2share failing to restart when elasticsearch is reused

### ISSUE THREAD

**Issue by bne86** [EUDAT-B2SHARE/b2share#1232]
When restarting b2share (maybe due to a update) with a existing elasticsearch service, b2share (at least the docker container running b2share) is not able to start with:

b2share_1        | PUT http://elasticsearch:9200/marc21-holdings-hd-v1.0.0 [status:400 request:0.030s]
b2share_1        | Traceback (most recent call last):
b2share_1        |   File "/usr/bin/b2share", line 11, in <module>
b2share_1        |     load_entry_point('b2share', 'console_scripts', 'b2share')()
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 716, in __call__
b2share_1        |     return self.main(*args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/flask/cli.py", line 345, in main
b2share_1        |     return AppGroup.main(self, *args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 696, in main
b2share_1        |     rv = self.invoke(ctx)
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 1060, in invoke
b2share_1        |     return _process_result(sub_ctx.command.invoke(sub_ctx))
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 1060, in invoke
b2share_1        |     return _process_result(sub_ctx.command.invoke(sub_ctx))
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 889, in invoke
b2share_1        |     return ctx.invoke(self.callback, **ctx.params)
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 534, in invoke
b2share_1        |     return callback(*args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/decorators.py", line 17, in new_func
b2share_1        |     return f(get_current_context(), *args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/flask/cli.py", line 229, in decorator
b2share_1        |     return __ctx.invoke(f, *args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/core.py", line 534, in invoke
b2share_1        |     return callback(*args, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/invenio_search/cli.py", line 61, in init
b2share_1        |     for name, response in bar:
b2share_1        |   File "/usr/lib/python3.4/site-packages/click/_termui_impl.py", line 257, in next
b2share_1        |     return next(self.iter)
b2share_1        |   File "/usr/lib/python3.4/site-packages/invenio_search/ext.py", line 158, in create
b2share_1        |     for result in _create(self.aliases):
b2share_1        |   File "/usr/lib/python3.4/site-packages/invenio_search/ext.py", line 148, in _create
b2share_1        |     for result in _create(value, alias=name):
b2share_1        |   File "/usr/lib/python3.4/site-packages/invenio_search/ext.py", line 148, in _create
b2share_1        |     for result in _create(value, alias=name):
b2share_1        |   File "/usr/lib/python3.4/site-packages/invenio_search/ext.py", line 145, in _create
b2share_1        |     ignore=ignore,
b2share_1        |   File "/usr/lib/python3.4/site-packages/elasticsearch/client/utils.py", line 69, in _wrapped
b2share_1        |     return func(*args, params=params, **kwargs)
b2share_1        |   File "/usr/lib/python3.4/site-packages/elasticsearch/client/indices.py", line 110, in create
b2share_1        |     params=params, body=body)
b2share_1        |   File "/usr/lib/python3.4/site-packages/elasticsearch/transport.py", line 327, in perform_request
b2share_1        |     status, headers, data = connection.perform_request(method, url, params, body, ignore=ignore, timeout=timeout)
b2share_1        |   File "/usr/lib/python3.4/site-packages/elasticsearch/connection/http_requests.py", line 84, in perform_request
b2share_1        |     self._raise_error(response.status_code, raw_data)
b2share_1        |   File "/usr/lib/python3.4/site-packages/elasticsearch/connection/base.py", line 113, in _raise_error
b2share_1        |     raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
b2share_1        | elasticsearch.exceptions.RequestError: TransportError(400, 'index_already_exists_exception', 'already exists')
dockerize_b2share_1 exited with code 1

Is this really something that should happen? Attaching a service to some level of persistency fails.

**Comment by emanueldima** [comment]
Adding a volume to hold the elasticsearch solved the issue for me:
````
elasticsearch:
    image: elasticsearch
    volumes:
        - /elasticsearchdata:/usr/share/elasticsearch/data
````

**Comment by emanueldima** [comment]
@bne86 If possible, please provide the exact sequence of docker operations that triggers this error.

**Comment by bne86** [comment]
I am not really able to provide the exact command order, but in general it should have been a simple 

> docker-compose up

from inside of the dockerize directory. I have added the volumes command, but with docker-compose up he should have used the old docker container and with this already the old data/old index. And with this with this, the problem of having a HTTP400 because of a already existing index should also still exist?

**Comment by emanueldima** [comment]
The current way to update the site is to reindex all records (a matter of seconds at the current level of usage). I don't know yet how to solve this issue, might be better solved in Invenio.

**Comment by nharraud** [comment]
This issue should be solved by #1402.

**Comment by emanueldima** [comment]
solved in v2.0.1 and above
